src/meshapi/models/node.py

```diff
@@ -92,9 +92,10 @@ class NodeType(models.TextChoices):
 
     def save(self, *args: Any, **kwargs: Any) -> None:
         if not self.network_number:
-            with advisory_lock("nn_assignment_lock"):
-                self.network_number = get_next_available_network_number()
-                super().save(*args, **kwargs)
+            with transaction.atomic():
```

Comment:
We need both `transaction.atomic()` and `advisory_lock()` because they protect different things. `get_next_available_network_number()` uses `select_for_update()`, which requires the creation of a database transaction in order to signal release the selected Install object at the end of the transaction block. 

However, we also need to protect the in-memory copy of the lowest NN from races also, since `get_next_available_network_number()` can be called concurrently. It is true that `transaction.atomic()` protects a given code block from being entered more than once, but since we're calling `get_next_available_network_number()` from multiple places, that protection doesn't apply, so we use `advisory_lock()` for supplemental protection